test/e2e_node/mirror_pod_readiness_test.go

```diff
@@ -0,0 +1,135 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"context"
+	"errors"
+	"os"
+	"time"
+
+	"fmt"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/kubernetes/test/e2e/framework"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+)
+
+var _ = SIGDescribe("MirrorPod", func() {
+	f := framework.NewDefaultFramework("mirror-pod")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelBaseline
+	ginkgo.Context("when create a mirror pod check readiness ", func() {
+		var ns, podPath, staticPodName, mirrorPodName string
+		ginkgo.BeforeEach(func(ctx context.Context) {
+			ns = f.Namespace.Name
+			staticPodName = "static-pod-" + string(uuid.NewUUID())
+			mirrorPodName = staticPodName + "-" + framework.TestContext.NodeName
+
+			podPath = framework.TestContext.KubeletConfig.StaticPodPath
+
+			ginkgo.By("create the static pod")
+			err := createStaticPodWithReadiness(podPath, staticPodName, ns,
+				imageutils.GetE2EImage(imageutils.Perl), v1.RestartPolicyAlways)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("wait for the mirror pod to be running")
+			gomega.Eventually(ctx, func(ctx context.Context) error {
+				return checkMirrorPodRunning(ctx, f.ClientSet, mirrorPodName, ns)
+			}, 2*time.Minute, time.Second*4).Should(gomega.BeNil())
+
+			ginkgo.By("wait for pod to be ready")
+			gomega.Eventually(ctx, func(ctx context.Context) error {
+				pod, err := f.ClientSet.CoreV1().Pods(ns).Get(ctx, mirrorPodName, metav1.GetOptions{})
+				if err != nil {
+					return fmt.Errorf("expected the mirror pod %q to be present: %v", mirrorPodName, err)
+				}
+				for _, c := range pod.Status.Conditions {
+					if c.Type == v1.PodReady && c.Status == v1.ConditionTrue {
+						return nil
+					}
+				}
+				return errors.New("pod needs to be ready")
+			}, 2*time.Minute, time.Second*4).Should(gomega.BeNil())
+		})
+		/*
+			Testname: Mirror Pod, readiness
+			Description: checks the readiness of a pod on termination
+		*/
+		ginkgo.It("should be unready with readiness probe failure on termination [NodeConformance]", func(ctx context.Context) {
+			// remove the static pod, pod should go unready
+			err := deleteStaticPod(podPath, staticPodName, ns)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("wait for the mirror pod to be not ready")
+			gomega.Eventually(ctx, func(ctx context.Context) error {
```

Comment:
This will pass as soon as the mirror pod becomes unready, but it sounds like we want the pod to be report unready during the entire termination. If the pod reports goes unready -> ready during termination, we won't catch it in the test.

Should this change to maybe `gomego.Consistently` after we first detect the pod goes unready? Or maybe even better we can setup a watch against the pod and validate all future updates ensure that pod reports unready. Here is example of that - https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/node_shutdown_linux_test.go#L218-L240